nengo_gui/components/pointer.py

```diff
@@ -43,16 +45,18 @@ def remove_nengo_objects(self, page):
 
     def gather_data(self, t, x):
         vocab = self.vocab_out
-        m = np.dot(vocab.vectors, x)
-        matches = [(mm, vocab.keys[i]) for i, mm in enumerate(m) if mm > 0.01]
+        key_similarity = np.dot(vocab.vectors, x)
+        matches = [(simi, vocab.keys[i]) for i, simi in enumerate(key_similarity) if simi > 0.01]
```

Comment:
`key_similarity` is a Numpy array. That should allow for something like
```python
over_threshold = key_similarities > 0.01
matches = zip(key_similarities[over_threshold], vocab.keys[over_threshold])
```
I would expect that to be slightly faster and for me it is easier to understand.

Same thing applies below to the pairs stuff.